### ISSUE TITLE
Allow code exports to be defined via a plugin

### ISSUE THREAD

**Issue by poorgeek** [SeleniumHQ/selenium-ide#840]
## 🚀 Feature Proposal

Allow plugins to define additional code export functionality rather than relying on the export functions available only from Selenium IDE.

## Motivation

When distributing a plugin with custom commands and events to an internal team, it would be nice to include the ability to export the test case using a code export that matches our defined language and requirements. This could be as simple as including additional formatting in the export, new languages/frameworks, etc...

The ability to specific code exports in a plugin would also allow for quicker iterating of code-exports for internal distribution rather than waiting on changes to be merged into the Selenium IDE project.

## Example

We have a few custom commands that we are packaging as part of a plugin. As part of that, I want to create a custom export that supports the old Selenese XML format as we've got a few processes that we don't have the ability to update to support the new Side JSON format. By distributing the code-export for Selenese with the custom commands, I can give our team a single additional plugin to install with everything they need to use the new Selenium IDE now.

**Comment by corevo** [comment]
Our plugin system already allows defining more export targets, you can consult the documentation on how to do so, and develop a plugin.

We have no plans on converting SIDE back to XML, since it will limit the new formats capabilities.

**Comment by poorgeek** [comment]
Sorry about that. I saw the Emitting Code, but must've skipped right over [Exporting Code](https://www.seleniumhq.org/selenium-ide/docs/en/plugins/code-export/) in the docs.
